creation/interfaces-and-networks.md

```diff
@@ -548,6 +548,51 @@ the interface should be configured as follows.
 > **Note:** The network CIDR can be configured in the pod network
 > section using the `vmNetworkCIDR` attribute.
 
+#### masquerade - IPv6 support
+It is currently experimental, but `masquerade` mode can be used in IPv6 clusters
+(not dual stack - IPv6 only).
+
+As with the IPv4 `masquerade` mode, the VM can be contacted using the pod's IP
+address - which will be an IPv6 one. Outgoing traffic is also "NAT'ed" to the
+pod's IPv6 address.
+
+Unlike in IPv4, the configuration of the IP address, default route, and DNS is
+not automatic; it should be configured via cloud init, as shown below:
+
+    kind: VM
+    spec:
+      domain:
+        devices:
+          disks:
+            - disk:
+              bus: virtio
+              name: cloudinitdisk
+          interfaces:
+            - name: red
+              masquerade: {} # connect using masquerade mode
+              ports:
+                - port: 80 # allow incoming traffic on port 80 to get into the virtual machine
+      networks:
+      - name: red
+        pod: {}
+      volumes:
+      - cloudInitNoCloud:
+          userData: |-
+            #!/bin/bash
+            echo "fedora" |passwd fedora --stdin
+            ip -6 addr add fd2e:f1fe:9490:a8ff::2/120 dev eth0
+            sleep 5
+            ip -6 route add default via fd2e:f1fe:9490:a8ff::1 src fd2e:f1fe:9490:a8ff::2
```

Comment:
we saw that sudo is needed right ?

Reply:
It is not needed.

I'm copying what's on the [masquerade example](https://github.com/kubevirt/kubevirt/blob/master/examples/vmi-masquerade.yaml#L43).

```
$ kubectl apply -f examples/vmi-masquerade.yaml    
virtualmachineinstance.kubevirt.io/vmi-masquerade created

$ kubectl get vmi vmi-masquerade                   
NAME             AGE   PHASE     IP            NODENAME
vmi-masquerade   41s   Running   10.244.1.63   node02

$ ./_out/cmd/virtctl/virtctl console vmi-masquerade
[fedora@vmi-masquerade ~]$ ip a show eth0
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc fq_codel state UP group default qlen 1000
    link/ether 02:00:00:42:36:4c brd ff:ff:ff:ff:ff:ff
    altname enp1s0
    inet 10.0.2.2/24 brd 10.0.2.255 scope global dynamic noprefixroute eth0
       valid_lft 86313541sec preferred_lft 86313541sec
    inet6 fd2e:f1fe:9490:a8ff::2/120 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::ff:fe42:364c/64 scope link 
       valid_lft forever preferred_lft forever
[fedora@vmi-masquerade ~]$ ip -6 r
::1 dev lo proto kernel metric 256 pref medium
fd2e:f1fe:9490:a8ff::/120 dev eth0 proto kernel metric 256 pref medium
fe80::/64 dev eth0 proto kernel metric 256 pref medium
default via fd2e:f1fe:9490:a8ff::1 dev eth0 src fd2e:f1fe:9490:a8ff::2 metric 1024 pref medium

```

Reply:
ok i c
just remember for sure that in the test we had to add it in the begining at least
without it it didnt work

up to you thx